src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,114 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double GROW_FACTOR = 1.5;
+    private int size;
+    private T[] dataArray;
+
+    public ArrayList() {
+        this.dataArray = (T[]) new Object[DEFAULT_CAPACITY];
```

Comment:
```suggestion
        dataArray = (T[]) new Object[DEFAULT_CAPACITY];
```
where do we need [`this`](https://stackoverflow.com/a/2411283)